src/Tasks/Microsoft.NET.Build.Tasks/GenerateToolsSettingsFile.cs

```diff
@@ -9,34 +9,96 @@ namespace Microsoft.NET.Build.Tasks
 {
     public class GenerateToolsSettingsFile : TaskBase
     {
-        // bump whenever the format changes such that it will break old consumers
-        private static readonly int _formatVersion = 1;
-
         [Required]
         public string EntryPointRelativePath { get; set; }
 
         [Required]
         public string CommandName { get; set; }
 
+        public string CommandRunner { get; set; }
+
+        public string RuntimeIdentifier { get; set; }
+
+        public string ToolPackageId { get; set; }
+
+        public string ToolPackageVersion { get; set; }
+
+        public ITaskItem[] ToolPackageRuntimeIdentifiers { get; set; }
+
         [Required]
         public string ToolsSettingsFilePath { get; set; }
 
         protected override void ExecuteCore()
         {
-            GenerateDocument(EntryPointRelativePath, CommandName).Save(ToolsSettingsFilePath);
+            GenerateDocument(EntryPointRelativePath, CommandName, CommandRunner, RuntimeIdentifier, ToolPackageId, ToolPackageVersion, ToolPackageRuntimeIdentifiers)
+                .Save(ToolsSettingsFilePath);
         }
 
-        internal static XDocument GenerateDocument(string entryPointRelativePath, string commandName)
+        internal static XDocument GenerateDocument(string entryPointRelativePath, string commandName, string commandRunner, string runtimeIdentifier,
+            string toolPackageId, string toolPackageVersion, ITaskItem[] toolPackageRuntimeIdentifiers)
         {
+            // Format version should bump whenever the format changes such that it will break old consumers
+            int formatVersion = 1;
+
+            if (string.IsNullOrEmpty(commandRunner))
+            {
+                commandRunner = "dotnet";
+            }
+
+            if (commandRunner != "dotnet")
+            {
+                formatVersion = 2;
```

Comment:
Can this be linked back to the other formatVersion in the CLI or does that not make sense, something like `const int LatestToolFormatVersion = 2;` ? Also, should we check for specific command runners

Reply:
I didn't do something like that since the version written depends on the features you use, so if you don't use new features we'll use the old format version and be backwards compatible.

I have now just added an error if you set the runner to something besides "dotnet" or "executable".